src/picongpu/include/particles/gasProfiles/FreeFormulaImpl.hpp

```diff
@@ -51,9 +51,15 @@ struct FreeFormulaImpl : public T_ParamClass
      */
     HDINLINE float_X operator()(const DataSpace<simDim>& totalCellOffset)
     {
-        const float_64 unit_length = UNIT_LENGTH;
+        DataSpace<simDim> position_SI = totalCellOffset;
+        cellSize_t cellSize_SI = cellSize;
+        for(unsigned int i = 0; i<simDim ; i++)
+        {
+            cellSize_SI *= UNIT_LENGTH;
```

Comment:
`cellSize_t` is of type [float_X](https://github.com/ComputationalRadiationPhysics/picongpu/blob/4c6c28c1b349abdda5b7e4178071d4a9c5aeab86/src/picongpu/include/simulation_defines/unitless/gridConfig.unitless#L35) and does not guarantee to hold a SI-scaled value.